CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKIconsRenderable.java

```diff
@@ -243,41 +413,37 @@ private static int LoadShaders(CVKDevice cvkDevice) {
+            if (vsBytes == null) {
+                vsBytes = LoadFileToDirectBuffer(CVKShaderPlaceHolder.class, "compiled/VertexIcon.vs.spv");
+                if (vsBytes == null) {
+                    CVKLOGGER.log(Level.SEVERE, "Failed to load precompiled CVKIconsRenderable shader: VertexIcon.vs");
+                    return CVK_ERROR_SHADER_COMPILATION;
+                }
             }
-            ByteBuffer gsBytes = LoadFileToDirectBuffer(CVKShaderPlaceHolder.class, "compiled/VertexIcon.gs.spv");
-            if (vsBytes.capacity() == 0) {
-                throw new RuntimeException("Failed to load compiled/SimpleIcon.gs.spv");
-            }            
-            ByteBuffer fsBytes = LoadFileToDirectBuffer(CVKShaderPlaceHolder.class, "compiled/VertexIcon.fs.spv");
-            if (vsBytes.capacity() == 0) {
-                throw new RuntimeException("Failed to load compiled/SimpleIcon.fs.spv");
-            }            
             
-            hVertexShader = CVKShaderUtils.createShaderModule(vsBytes, cvkDevice.GetDevice());
-            if (hVertexShader == VK_NULL_HANDLE) {
-                throw new RuntimeException("Failed to create shader from VertexIcon.vs.spv bytes");
-            }            
-            hGeometryShader = CVKShaderUtils.createShaderModule(gsBytes, cvkDevice.GetDevice());
-            if (hGeometryShader == VK_NULL_HANDLE) {
-                throw new RuntimeException("Failed to create shader from VertexIcon.gs.spv bytes");
-            }             
-            hFragmentShader = CVKShaderUtils.createShaderModule(fsBytes, cvkDevice.GetDevice());            
-            if (hFragmentShader == VK_NULL_HANDLE) {
-                throw new RuntimeException("Failed to create shader from VertexIcon.fs.spv bytes");
-            }      
+            if (gsBytes == null) {
+                gsBytes = LoadFileToDirectBuffer(CVKShaderPlaceHolder.class, "compiled/VertexIcon.gs.spv");
+                if (gsBytes == null) {
+                    CVKLOGGER.log(Level.SEVERE, "Failed to load precompiled CVKIconsRenderable shader: VertexIcon.gs");
+                    return CVK_ERROR_SHADER_COMPILATION;
+                }
+            }
             
-            MemoryUtil.memFree(vsBytes);
-            MemoryUtil.memFree(gsBytes);
-            MemoryUtil.memFree(fsBytes);
+            if (fsBytes == null) {
+                fsBytes = LoadFileToDirectBuffer(CVKShaderPlaceHolder.class, "compiled/VertexIcon.fs.spv");
+                if (fsBytes == null) {
+                    CVKLOGGER.log(Level.SEVERE, "Failed to load precompiled CVKIconsRenderable shader: VertexIcon.fs");
+                    return CVK_ERROR_SHADER_COMPILATION;
+                }
+            }
+         
         } catch (IOException e) {
-            //TODO_TT
+            CVKLOGGER.log(Level.SEVERE, "Failed to compile FPSRenderable shaders: {0}", e.toString());
```

Comment:
FPS -> Icon

Reply:
Eh this will be getting moved anyway.  Happy to skip for now?